open-metadata-implementation/common-services/generic-handlers/src/main/java/org/odpi/openmetadata/commonservices/generichandlers/SchemaAttributeHandler.java

```diff
@@ -1865,6 +1874,169 @@ public void   updateSchemaAttribute(String               userId,
+        {
+            attributeTypeName = typeName;
+        }
+
+        EntityDetail schemaAttributeEntity = this.getEntityFromRepository(userId,
+                                                                          schemaAttributeGUID,
+                                                                          schemaAttributeGUIDParameterName,
+                                                                          attributeTypeName,
+                                                                          null,
+                                                                          null,
+                                                                          forLineage,
+                                                                          forDuplicateProcessing,
+                                                                          effectiveTime,
+                                                                          methodName);
+
+        if (schemaAttributeEntity != null)
+        {
+            InstanceProperties instanceProperties = schemaAttributeBuilder.getInstanceProperties(methodName);
+
+            this.updateBeanInRepository(userId,
+                                        externalSourceGUID,
+                                        externalSourceName,
+                                        schemaAttributeGUID,
+                                        schemaAttributeGUIDParameterName,
+                                        OpenMetadataAPIMapper.SCHEMA_ATTRIBUTE_TYPE_GUID,
+                                        OpenMetadataAPIMapper.SCHEMA_ATTRIBUTE_TYPE_NAME,
+                                        forLineage,
+                                        forDuplicateProcessing,
+                                        supportedZones,
+                                        instanceProperties,
+                                        true,
+                                        effectiveTime,
+                                        methodName);
+
+            SchemaTypeBuilder schemaTypeBuilder = schemaAttributeBuilder.getSchemaTypeBuilder();
+            if (schemaTypeBuilder != null)
+            {
+
+                /*
+                 * The formula is set if the schema attribute is derived. Need to test the merge semantics.
+                 */
+                String formula = schemaTypeBuilder.getFormula();
+                if (formula == null)
+                {
+                    // if we have no formula requested and we are not a merge, any existing
+                    // calculated value classification should be cleared
+                    if (!isMergeUpdate)
+                    {
+                        List<Classification> classifications = schemaAttributeEntity.getClassifications();
+                        if (classifications != null && classifications.size() >0)
+                        {
+                            for (Classification classification : classifications)
```

Comment:
Why reinvent this loop rather than using repositoryHelper.getClassificationFromEntity()?

Reply:
thanks - changed